samples/MicStream.js

```diff
@@ -0,0 +1,378 @@
+  console.log(`resolved filename ${filename}`);
+  const audio = {
+    content: fs.readFileSync(filename).toString('base64'),
+  };
+
+  const request = {
+    config: config,
+    audio: audio,
+  };
+
+  // Detects speech in the audio file
+  client
+    .recognize(request)
+    .then(data => {
+      const response = data[0];
+      const transcription = response.results
+        .map(result => result.alternatives[0].transcript)
+        .join('\n');
+      console.log(`wavFileToText: Transcription; \n`, transcription);
+      wavtranscription = transcription;
+      const confidence = response.results
+        .map(result => result.alternatives[0].confidence)
+        .join('\n');
+      console.log(`wavFileToText: Confidence; \n`, confidence);
+      eventEmitter.emit(`wavFileToText: complete`);
+    })
+    .catch(err => {
+      console.error('ERROR:', err);
+    });
+  // [END wavFileToText]
+}
+
+function test_1(
+  targetfile,
+  recordingtime,
+  encoding,
+  sampleRateHertz,
+  languageCode
+) {
+  const fs = require('fs');
+  if (targetfile.substr(targetfile.length - 4) !== `.wav`)
+    targetfile = targetfile + `.wav`;
+  eventEmitter.emit(`test_1: Started`);
+  fs.access(targetfile, fs.constants.W_OK, err => {
+    console.log(`${targetfile} ${err ? 'is not writable' : 'is writable'}`);
+  });
+  var streamRes = micStreamRecognize(
+    targetfile,
+    recordingtime,
+    encoding,
+    sampleRateHertz,
+    languageCode
+  );
+  var test1trans = [];
+  eventEmitter.on(`micStreamRecognize: complete`, () => {
+    wavFileToText(targetfile, encoding, sampleRateHertz, languageCode);
+    for (var i in streamRes) {
+      if (streamRes[i].length < 3) test1trans.push(`\n` + streamRes[i][0]);
+    }
+    console.log(`test_1: Streaming result(s); ${test1trans}`);
+    console.log(`test_1: Complete!  File at: ${targetfile}`);
+  });
+  eventEmitter.on(`wavFileToText: complete`, () => {
+    compareLiveStreamtoWavFile(wavtranscription, streamtranscription);
+  });
+}
+
+//globals to assist with compareLiveStreamtoWavFile test (below)
+var wavtranscription;
+var streamtranscription;
+
+//Simple test function that outputs to console results of comparing stream to .wav translation
+function compareLiveStreamtoWavFile(wavtranscription, streamtranscription) {
+  const wav = wavtranscription;
+  const stream = streamtranscription;
+  if (wav !== null && stream !== null) {
+    console.log(
+      `Is .Wav speech to text is same as Streamed speech to text?: ${(
+        wav === stream
+      ).toString()}`
+    );
+  }
+  console.log(`stream speech to text is: ${stream}`);
+  console.log(`   wav speech to text is: ${wav}`);
+}
+
+require(`yargs`)
+  .demand(1)
+  .command(
+    `run`,
+    `Runs complete streaming test: \n 1. streams from microphone, outputting streamed speech-to-text to console, \n 2. records that streamed sample to resources folder, \n 3. runs speech-to-text on that local file to compare to step 1.`,
```

Comment:
Cool idea, but not part of this sample

Let's remove the code that saves to a .wav file, it's out of scope (and it's how people usually use Speech-to-Text, see the existing samples in this directory)